src/main/scala/is/hail/expr/Type.scala

```diff
@@ -115,6 +115,13 @@ sealed abstract class Type extends Serializable { self =>
       (TStruct.empty, a => null)
   }
 
+  def unsafeInsert(typeToInsert: Type, path: List[String]): (Type, UnsafeInserter) = {
```

Comment:
The existence of this basically lets us overwrite non-struct members, right? For example, `TStruct("a" -> Int).unsafeInsert(Int, List("a", "b"))` blows away whatever value was stored at `"a"`.